src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -262,63 +344,103 @@ class npc_mograine : public CreatureScript
-                    return 0;
-                default:
-                    if(mograine)
-                        mograine->DespawnOrUnsummon(0);
-                    return 0;
+            case EVENT_MOGRAINE_FACING_PLAYER:
+                me->SetFacingToObject(playerWhoStartedAshbringer);
+                events.ScheduleEvent(EVENT_MOGRAINE_KNEEL, 1s, 3s);
+                break;
+            case EVENT_MOGRAINE_KNEEL:
+                me->SetSheath(SHEATH_STATE_UNARMED);
+                me->SetStandState(UNIT_STAND_STATE_KNEEL);
+                events.ScheduleEvent(EVENT_MOGRAINE_EMOTE_TALK3, 1s, 2s);
+                break;
+            case EVENT_MOGRAINE_EMOTE_TALK3:
+                me->HandleEmoteCommand(EMOTE_ONESHOT_TALK);
+                me->AI()->Talk(SAY_MO_AB_TALK3, playerWhoStartedAshbringer,43ms);
+                break;
+            case EVENT_SUMMONED_HIGHLORD_MOGRAINE:
+                if (Creature* summonedMograine = me->SummonCreature(NPC_HIGHLORD_MOGRAINE, 1033.4642f, 1399.1022f, 27.337427f, 6.257956981658935546f, TEMPSUMMON_TIMED_DESPAWN, 120000))
+                {
+                    summonedMograine->SetFaction(FACTION_FRIENDLY);
+                    summonedMograine->SetUInt32Value(UNIT_VIRTUAL_ITEM_SLOT_ID + 0, EQUIP_UNEQUIP);
+                    summonedMograine->SetSpeed(MOVE_WALK, 1.0f);
+                    summonedMograine->SetDisplayId(MODEL_HIGHLORD_MOGRAINE);
+                    summonedMograine->setActive(true);
+                    summonedMograine->CastSpell(summonedMograine, SPELL_MOGRAINE_COMETH_DND, false);//Sniffing data shows the use of this spell transformation, but the dispersion effect of this spell is not seen in the video
+                }
+                events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_MOVE_STOP, 48546ms);
+                break;
+            case EVENT_HIGHLORD_MOGRAINE_MOVE_STOP:
+                summonedMograine->StopMovingOnCurrentPos();
+                summonedMograine->HandleEmoteCommand(EMOTE_ONESHOT_POINT);
+                summonedMograine->AI()->Talk(SAY_HM_AB_TALK0, 214ms);
+                events.ScheduleEvent(EVENT_MOGRAINE_FACING_HIGHLORD_MOGRAINE, 3024ms);
+                break;
+            case EVENT_MOGRAINE_FACING_HIGHLORD_MOGRAINE:
+                me->SetFacingToObject(summonedMograine);
+                events.ScheduleEvent(EVENT_MOGRAINE_UNIT_STAND_STATE_STAND, 403ms);
+                break;
+            case EVENT_MOGRAINE_UNIT_STAND_STATE_STAND:
+                me->SetStandState(UNIT_STAND_STATE_STAND);
+                me->SetSheath(SHEATH_STATE_MELEE);
+                events.ScheduleEvent(EVENT_MOGRAINE_EMOTE_TALK4, 809ms);
+                break;
+            case EVENT_MOGRAINE_EMOTE_TALK4:
+                me->HandleEmoteCommand(EMOTE_ONESHOT_QUESTION);
+                me->AI()->Talk(SAY_MO_AB_TALK4, 209ms);
+                events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_EMOTE_TALK, 4636ms);
```

Comment:
This is a bit overkill.

Remember there is server delay too so sniffed values need some tuning.

This is pribably 4600ms
The lne a above 200ms

I would say anything 0-99ms is probably just serverdelay in sniffs, so round them to the nearest hundred

Reply:
I asked before. The reply was to keep the sniffed data, so I'm going to keep the original time

Reply:
Does time elsewhere in the code need to be rounded

Reply:
everywhere